### ISSUE TITLE
Workaround for "Failed to update dependencies: Maximum retries exceeded"

### ISSUE THREAD

**Issue by amsoedal** [microsoft/mindaro#32]
**The issue**
Some of the time, getting dependencies for Bridge to Kubernetes in VS Code may fail with this error. 

![image](https://user-images.githubusercontent.com/13313748/93949536-b6a9f480-fcf5-11ea-9d8e-fc6749da17de.png)
If the issue reproduces consistently after hitting the retry button, a manual workaround is the best way to proceed. 

**The workaround**
1. From a browser, visit https://aka.ms/lpk-lks. You should see a JSON blob with a version number, and three platform-dependent download links.
2. Open the download link for your platform from step 1 to acquire the binaries.
3. Unzip and move the contents to the Visual Studio Code extension folder.  You may need to create the directory if it does not exist.
On Windows: %UserProfile%\AppData\Roaming\Code\User\globalStorage\mindaro.mindaro\file-downloader-downloads\binaries
On MacOS: "/Users/<username>/Library/Application Support/Code/User/globalStorage/mindaro.mindaro/file-downloader-downloads/binaries"
On Linux: /home/<username>/.config/Code/User/globalStorage/mindaro.mindaro/file-downloader-downloads/binaries
4. To check that the hierarchy is correct, you should now see folders called "kubectl" and "EndpointManager" along with several other files side-by-side under the /binaries/ directory. 
5. Open VS Code again.  The extension should now be able to find and execute its dependencies.

**Long-term**
We are currently at work trying to understand why this error happens and to improve the experience! Thank you for your patience.

**Comment by getmr** [comment]
Failed to update dependencies: XHR failed

**Comment by sammym1982** [comment]
I have another flavor of same error . But these workaround does not seems to work :(

![image](https://user-images.githubusercontent.com/11453064/100579876-d9c6c680-3299-11eb-8c3b-ebc73ef2a10a.png)

**Comment by daniv-msft** [comment]
@sammym1982 Thank you for reporting this, and sorry you have this issue. Could you please try to terminate the EndpointManager process manually (if you're on Windows, you should see this right in the Task Manager) and restart VS Code?
This is the process we start with elevated permissions when using Bridge to Kubernetes. We have some logic to automatically kill it when updating dependencies, but it seems that this logic didn't work as expected in your case.

**Comment by sammym1982** [comment]
Yes that was it. Thanks. One other thing to note when this process was running one of the logs file kept growing and in my case went upto almost 500GB which made my c drive go out of space.

**Comment by daniv-msft** [comment]
Thanks for your reply. Do you remember which log file it was? This is not something we caught earlier, and is **very** worrying so any help to track this would be appreciated.

**Comment by SVronskiy** [comment]
Got problem:
```
Failed to update dependencies: EPERM: operation not permitted, rename 'c:\Users\svron\AppData\Roaming\Code\User\globalStorage\mindaro.mindaro\file-downloader-downloads\ad5081f1-aa9e-4add-a9ad-3f768504a8f7' -> 'c:\Users\svron\AppData\Roaming\Code\User\globalStorage\mindaro.mindaro\file-downloader-downloads\binaries'
Source: Bridge to Kubrnetes
```
There is no EndpointManager process in the Task Manager.
Manually renamed folder 'ad5081f1-aa9e-4add-a9ad-3f768504a8f7' to 'binaries' and this resolved my problem.

**Comment by daniv-msft** [comment]
Thanks @SVronskiy! We're aware of this issue and are investigating it, but we don't have a good solution yet. This might be a race condition where we try to rename the folder after downloading whereas a handle is still kept on it.
Adding @rakeshvanga for FYI.

**Comment by Alkhir** [comment]
https://discuss.python.org/t/pep-596-python-3-9-release-schedule-doubling-the-release-cadence/1828

**Comment by greenie-msft** [comment]
Hi everyone, 

We have updated our code to be more resilient when downloading dependencies. Please let us know if you are still experiencing this issue.  Keeping this thread open for now.

Thank you

**Comment by chudytom** [comment]
It still doesn't work for me when trying to run Kubernetes Bridge. It doesn't work on Windows nor in WSL. And the error I'm getting is
"Failed to update dependencies: Maximum number of retries exceeded. The operation 'HttpRequestHandler.get' failed with error: unable to get issuer certificate. Technical details: {"message":"unable to get issuer certificate","name":"Error","stack":"Error: unable to get issuer certificate\n\tat TLSSocket.onConnectSecure (_tls_wrap.js:1497:34)\n\tat TLSSocket.emit (events.js:315:20)\n\tat TLSSocket._finishInit (_tls_wrap.js:932:8)\n\tat TLSWrap.ssl.onhandshakedone (_tls_wrap.js:706:12)","config":{"url":"https://bridgetokubernetes.azureedge.net/zip/20210517.1/lpk-linux.zip","method":"get","headers":{"Accept":"application/json, text/plain, */*","User-Agent":"axios/0.21.1"},"proxy":false,"transformRequest":[null],"transformResponse":[null],"timeout":5000,"responseType":"stream","xsrfCookieName":"XSRF-TOKEN","xsrfHeaderName":"X-XSRF-TOKEN","maxContentLength":-1,"maxBodyLength":-1},"code":"UNABLE_TO_GET_ISSUER_CERT"}"

**Comment by pragyamehta** [comment]
Hi @chudytom It seems you are unable to connect to our storage account where we host our binaries. Do you have any network policies that would prevent you from downloading our bits?
To get unblocked, you can try downloading our binaries manually from https://bridgetokubernetes.azureedge.net/zip/20210517.1/lpk-linux.zip

Once you download, you can unzip them and set the environment variable BRIDGE_BUILD_PATH to the binaries. Please let me know if this unblocks you. Thanks!

**Comment by chudytom** [comment]
Hi @pragyamehta Thanks a lot for the promtp response.

I tested the solution you proposed, downloaded the binaries manually put them in $HOME/bin/lpk-linux and set this location as BRIDGE_BUILD_PATH env var. 
The workaround worked very nicely. Thank you very much for your help 😄 
Before I tried to install the binaries manually on Windows and it didn't work. Is there a way to have it running on Windows as some developers on our side don't use WSL? Maybe we're able to ublock something specific in our firewal as indeed I am running behind corporate network policies. I tried to investigate what gets blocked specifically but still can't resolve it. 
What FQDN does the extension use under the hood to download the binaries?
What is the location of the binaries after they are downloaded? (Windows)
Do you have some other ideas how to investigate it?

**Comment by rafeeqg** [comment]
Hi,
Can someone help with this please? I've tried to uninstall/re-install but nothing seems to be working. I am using VSCode on a Windows 10 host
![image](https://user-images.githubusercontent.com/19571551/122729953-a0739980-d271-11eb-813b-d0bdb16cc1b5.png)
Thanks!

**Comment by amsoedal** [comment]
Hi @rafeeqg, thanks for reporting this issue. To get unblocked, did you try the steps from the first issue (downloading from the link and putting the dependencies in the folder)?

**Comment by rafeeqg** [comment]
Hi @amsoedal , yes, I did try the steps given in your post however, that didnt resolve the issue. 
Thanks!

**Comment by amsoedal** [comment]
@rafeeqg OK, thanks for giving that a go. Let's try this instead: download the zip linked in my post, extract it to a location (not the folder that I mentioned), and before you launch VS Code set this environment variable. BRIDGE_BUILD_PATH=<place where you extracted the zip>. Let me know if that changes anything.

**Comment by TinLe** [comment]
Version: 1.59.0
Commit: 379476f0e13988d90fab105c5c19e7abc8b1dea8
Date: 2021-08-04T23:14:40.191Z
Electron: 13.1.7
Chrome: 91.0.4472.124
Node.js: 14.16.0
V8: 9.1.269.36-electron.0
OS: Darwin x64 20.6.0

I am running into same errors.  Followed the above for manual install.  D/l the zip file and getting bad CRC errors while unzip it.  I've d/l twice already and same errors each time.

`:~/Library/Application Support/Code/User/globalStorage/mindaro.mindaro/file-downloader-downloads/binaries)-- unzip -o ~/Downloads/vsc/lpk-osx.zip
Archive:  /Users/tile/Downloads/vsc/lpk-osx.zip
  inflating: Autofac.dll
  inflating: AutoMapper.dll
  inflating: BouncyCastle.Crypto.dll
  inflating: dsc
  inflating: dsc.deps.json
  inflating: dsc.dll
  inflating: dsc.runtimeconfig.json
  inflating: EndpointManager/Autofac.dll
  inflating: EndpointManager/Autofac.Extensions.DependencyInjection.dll
  inflating: EndpointManager/AutoMapper.dll
  inflating: EndpointManager/BouncyCastle.Crypto.dll
  inflating: EndpointManager/EndpointManager
  inflating: EndpointManager/EndpointManager.deps.json
  inflating: EndpointManager/EndpointManager.dll
  inflating: EndpointManager/EndpointManager.runtimeconfig.json
  inflating: EndpointManager/Fractions.dll
  inflating: EndpointManager/KubernetesClient.dll
  inflating: EndpointManager/libclrjit.dylib
  inflating: EndpointManager/libcoreclr.dylib
  error:  invalid compressed data to inflate
  inflating: EndpointManager/libdbgshim.dylib
  inflating: EndpointManager/libhostfxr.dylib
  inflating: EndpointManager/libhostpolicy.dylib
  inflating: EndpointManager/libmscordaccore.dylib
  inflating: EndpointManager/libmscordbi.dylib
  inflating: EndpointManager/MessagePack.dll
  inflating: EndpointManager/Microsoft.ApplicationInsights.dll
  inflating: EndpointManager/Microsoft.AspNetCore.Antiforgery.dll
  inflating: EndpointManager/Microsoft.AspNetCore.Authentication.Abstractions.dll
  inflating: EndpointManager/Microsoft.AspNetCore.Authentication.Cookies.dll
  inflating: EndpointManager/Microsoft.AspNetCore.Authentication.Core.dll
  inflating: EndpointManager/Microsoft.AspNetCore.Authentication.dll
  inflating: EndpointManager/Microsoft.AspNetCore.Authentication.JwtBearer.dll
  inflating: EndpointManager/Microsoft.AspNetCore.Authentication.OAuth.dll
  inflating: EndpointManager/Microsoft.AspNetCore.Authorization.dll
  inflating: EndpointManager/Microsoft.AspNetCore.Authorization.Policy.dll
  inflating: EndpointManager/Microsoft.AspNetCore.Components.Authorization.dll
  inflating: EndpointManager/Microsoft.AspNetCore.Components.dll
  inflating: EndpointManager/Microsoft.AspNetCore.Components.Forms.dll
  inflating: EndpointManager/Microsoft.AspNetCore.Components.Server.dll
  inflating: EndpointManager/Microsoft.AspNetCore.Components.Web.dll
  inflating: EndpointManager/Microsoft.AspNetCore.Connections.Abstractions.dll
  inflating: EndpointManager/Microsoft.AspNetCore.CookiePolicy.dll
  inflating: EndpointManager/Microsoft.AspNetCore.Cors.dll
  inflating: EndpointManager/Microsoft.AspNetCore.Cryptography.Internal.dll
  inflating: EndpointManager/Microsoft.AspNetCore.Cryptography.KeyDerivation.dll
  inflating: EndpointManager/Microsoft.AspNetCore.DataProtection.Abstractions.dll
  inflating: EndpointManager/Microsoft.AspNetCore.DataProtection.dll
  inflating: EndpointManager/Microsoft.AspNetCore.DataProtection.Extensions.dll
  inflating: EndpointManager/Microsoft.AspNetCore.Diagnostics.Abstractions.dll
  inflating: EndpointManager/Microsoft.AspNetCore.Diagnostics.dll
  inflating: EndpointManager/Microsoft.AspNetCore.Diagnostics.HealthChecks.dll
  inflating: EndpointManager/Microsoft.AspNetCore.dll
  inflating: EndpointManager/Microsoft.AspNetCore.HostFiltering.dll
  inflating: EndpointManager/Microsoft.AspNetCore.Hosting.Abstractions.dll
  inflating: EndpointManager/Microsoft.AspNetCore.Hosting.dll
  inflating: EndpointManager/Microsoft.AspNetCore.Hosting.Server.Abstractions.dll
  inflating: EndpointManager/Microsoft.AspNetCore.Html.Abstractions.dll
  inflating: EndpointManager/Microsoft.AspNetCore.Http.Abstractions.dll
  inflating: EndpointManager/Microsoft.AspNetCore.Http.Connections.Client.dll
  inflating: EndpointManager/Microsoft.AspNetCore.Http.Connections.Common.dll
  inflating: EndpointManager/Microsoft.AspNetCore.Http.Connections.dll
  inflating: EndpointManager/Microsoft.AspNetCore.Http.dll
  inflating: EndpointManager/Microsoft.AspNetCore.Http.Extensions.dll
  inflating: EndpointManager/Microsoft.AspNetCore.Http.Features.dll
  inflating: EndpointManager/Microsoft.AspNetCore.HttpOverrides.dll
  inflating: EndpointManager/Microsoft.AspNetCore.HttpsPolicy.dll
  inflating: EndpointManager/Microsoft.AspNetCore.Identity.dll
  inflating: EndpointManager/Microsoft.AspNetCore.JsonPatch.dll
  inflating: EndpointManager/Microsoft.AspNetCore.Localization.dll
  inflating: EndpointManager/Microsoft.AspNetCore.Localization.Routing.dll
  inflating: EndpointManager/Microsoft.AspNetCore.Metadata.dll
  inflating: EndpointManager/Microsoft.AspNetCore.Mvc.Abstractions.dll
  inflating: EndpointManager/Microsoft.AspNetCore.Mvc.ApiExplorer.dll
  inflating: EndpointManager/Microsoft.AspNetCore.Mvc.Core.dll   bad CRC f01082f4  (should be 4740a4c2)
  inflating: EndpointManager/Microsoft.AspNetCore.Mvc.Cors.dll
  inflating: EndpointManager/Microsoft.AspNetCore.Mvc.DataAnnotations.dll
  inflating: EndpointManager/Microsoft.AspNetCore.Mvc.dll
  inflating: EndpointManager/Microsoft.AspNetCore.Mvc.Formatters.Json.dll
  inflating: EndpointManager/Microsoft.AspNetCore.Mvc.Formatters.Xml.dll
  inflating: EndpointManager/Microsoft.AspNetCore.Mvc.Localization.dll
  inflating: EndpointManager/Microsoft.AspNetCore.Mvc.NewtonsoftJson.dll
  inflating: EndpointManager/Microsoft.AspNetCore.Mvc.Razor.dll
  inflating: EndpointManager/Microsoft.AspNetCore.Mvc.Razor.Extensions.dll
  inflating: EndpointManager/Microsoft.AspNetCore.Mvc.RazorPages.dll
  inflating: EndpointManager/Microsoft.AspNetCore.Mvc.TagHelpers.dll
  inflating: EndpointManager/Microsoft.AspNetCore.Mvc.ViewFeatures.dll
  inflating: EndpointManager/Microsoft.AspNetCore.Razor.dll
  inflating: EndpointManager/Microsoft.AspNetCore.Razor.Language.dll
  inflating: EndpointManager/Microsoft.AspNetCore.Razor.Runtime.dll
  inflating: EndpointManager/Microsoft.AspNetCore.ResponseCaching.Abstractions.dll
  inflating: EndpointManager/Microsoft.AspNetCore.ResponseCaching.dll
  inflating: EndpointManager/Microsoft.AspNetCore.ResponseCompression.dll
  inflating: EndpointManager/Microsoft.AspNetCore.Rewrite.dll
  inflating: EndpointManager/Microsoft.AspNetCore.Routing.Abstractions.dll
  inflating: EndpointManager/Microsoft.AspNetCore.Routing.dll
  inflating: EndpointManager/Microsoft.AspNetCore.Server.HttpSys.dll
  inflating: EndpointManager/Microsoft.AspNetCore.Server.IIS.dll
  error:  invalid compressed data to inflate
  inflating: EndpointManager/Microsoft.AspNetCore.Server.IISIntegration.dll
  inflating: EndpointManager/Microsoft.AspNetCore.Server.Kestrel.Core.dll
  inflating: EndpointManager/Microsoft.AspNetCore.Server.Kestrel.dll
  inflating: EndpointManager/Microsoft.AspNetCore.Server.Kestrel.Https.dll
  inflating: EndpointManager/Microsoft.AspNetCore.Server.Kestrel.Transport.Abstractions.dll
  inflating: EndpointManager/Microsoft.AspNetCore.Server.Kestrel.Transport.Sockets.dll
  inflating: EndpointManager/Microsoft.AspNetCore.Session.dll
  inflating: EndpointManager/Microsoft.AspNetCore.SignalR.Client.Core.dll   bad CRC 41b03617  (should be 8ffb92df)
file #107:  bad zipfile offset (local header sig):  16421675
file #108:  bad zipfile offset (local header sig):  16431383
file #109:  bad zipfile offset (local header sig):  16466589
  inflating: EndpointManager/Microsoft.AspNetCore.SignalR.dll
  inflating: EndpointManager/Microsoft.AspNetCore.SignalR.Protocols.Json.dll
  inflating: EndpointManager/Microsoft.AspNetCore.SignalR.Protocols.MessagePack.dll
  inflating: EndpointManager/Microsoft.AspNetCore.StaticFiles.dll
  inflating: EndpointManager/Microsoft.AspNetCore.WebSockets.dll
  inflating: EndpointManager/Microsoft.AspNetCore.WebUtilities.dll
  inflating: EndpointManager/Microsoft.Azure.Management.ResourceManager.Fluent.dll
  inflating: EndpointManager/Microsoft.CodeAnalysis.CSharp.dll
  inflating: EndpointManager/Microsoft.CodeAnalysis.dll
  inflating: EndpointManager/Microsoft.CodeAnalysis.Razor.dll
  inflating: EndpointManager/Microsoft.CSharp.dll
  inflating: EndpointManager/Microsoft.DevSpaces.Common.dll
  inflating: EndpointManager/Microsoft.DevSpaces.Common.Json.dll
  inflating: EndpointManager/Microsoft.DotNet.PlatformAbstractions.dll
  inflating: EndpointManager/Microsoft.Extensions.Caching.Abstractions.dll
  inflating: EndpointManager/Microsoft.Extensions.Caching.Memory.dll
  inflating: EndpointManager/Microsoft.Extensions.CommandLineUtils.dll
  inflating: EndpointManager/Microsoft.Extensions.Configuration.Abstractions.dll
  inflating: EndpointManager/Microsoft.Extensions.Configuration.Binder.dll
  inflating: EndpointManager/Microsoft.Extensions.Configuration.CommandLine.dll
  inflating: EndpointManager/Microsoft.Extensions.Configuration.dll
  inflating: EndpointManager/Microsoft.Extensions.Configuration.EnvironmentVariables.dll
  inflating: EndpointManager/Microsoft.Extensions.Configuration.FileExtensions.dll
  inflating: EndpointManager/Microsoft.Extensions.Configuration.Ini.dll
  inflating: EndpointManager/Microsoft.Extensions.Configuration.Json.dll
  inflating: EndpointManager/Microsoft.Extensions.Configuration.KeyPerFile.dll
  inflating: EndpointManager/Microsoft.Extensions.Configuration.UserSecrets.dll
  inflating: EndpointManager/Microsoft.Extensions.Configuration.Xml.dll
  inflating: EndpointManager/Microsoft.Extensions.DependencyInjection.Abstractions.dll
  inflating: EndpointManager/Microsoft.Extensions.DependencyInjection.dll
  inflating: EndpointManager/Microsoft.Extensions.DependencyModel.dll
  inflating: EndpointManager/Microsoft.Extensions.Diagnostics.HealthChecks.Abstractions.dll
  inflating: EndpointManager/Microsoft.Extensions.Diagnostics.HealthChecks.dll
  inflating: EndpointManager/Microsoft.Extensions.FileProviders.Abstractions.dll
  inflating: EndpointManager/Microsoft.Extensions.FileProviders.Composite.dll
  inflating: EndpointManager/Microsoft.Extensions.FileProviders.Embedded.dll
  inflating: EndpointManager/Microsoft.Extensions.FileProviders.Physical.dll
  inflating: EndpointManager/Microsoft.Extensions.FileSystemGlobbing.dll
  inflating: EndpointManager/Microsoft.Extensions.Hosting.Abstractions.dll
  inflating: EndpointManager/Microsoft.Extensions.Hosting.dll
  inflating: EndpointManager/Microsoft.Extensions.Http.dll
  inflating: EndpointManager/Microsoft.Extensions.Identity.Core.dll
  inflating: EndpointManager/Microsoft.Extensions.Identity.Stores.dll
  inflating: EndpointManager/Microsoft.Extensions.Localization.Abstractions.dll
  inflating: EndpointManager/Microsoft.Extensions.Localization.dll
  inflating: EndpointManager/Microsoft.Extensions.Logging.Abstractions.dll
  inflating: EndpointManager/Microsoft.Extensions.Logging.Configuration.dll
  inflating: EndpointManager/Microsoft.Extensions.Logging.Console.dll
  inflating: EndpointManager/Microsoft.Extensions.Logging.Debug.dll
  inflating: EndpointManager/Microsoft.Extensions.Logging.dll
  inflating: EndpointManager/Microsoft.Extensions.Logging.EventLog.dll
  inflating: EndpointManager/Microsoft.Extensions.Logging.EventSource.dll
  inflating: EndpointManager/Microsoft.Extensions.Logging.TraceSource.dll
  inflating: EndpointManager/Microsoft.Extensions.ObjectPool.dll
  inflating: EndpointManager/Microsoft.Extensions.Options.ConfigurationExtensions.dll
  inflating: EndpointManager/Microsoft.Extensions.Options.DataAnnotations.dll
  inflating: EndpointManager/Microsoft.Extensions.Options.dll
  inflating: EndpointManager/Microsoft.Extensions.Primitives.dll
  inflating: EndpointManager/Microsoft.Extensions.WebEncoders.dll
  inflating: EndpointManager/Microsoft.IdentityModel.Clients.ActiveDirectory.dll
  inflating: EndpointManager/Microsoft.IdentityModel.Clients.ActiveDirectory.Platform.dll
  inflating: EndpointManager/Microsoft.IdentityModel.JsonWebTokens.dll
  inflating: EndpointManager/Microsoft.IdentityModel.Logging.dll
  inflating: EndpointManager/Microsoft.IdentityModel.Protocols.dll
  inflating: EndpointManager/Microsoft.IdentityModel.Protocols.OpenIdConnect.dll
  inflating: EndpointManager/Microsoft.IdentityModel.Tokens.dll
  inflating: EndpointManager/Microsoft.JSInterop.dll
  inflating: EndpointManager/Microsoft.Net.Http.Headers.dll
  inflating: EndpointManager/Microsoft.Rest.ClientRuntime.Azure.Authentication.dll
  inflating: EndpointManager/Microsoft.Rest.ClientRuntime.Azure.dll
  inflating: EndpointManager/Microsoft.Rest.ClientRuntime.dll
  inflating: EndpointManager/Microsoft.VisualBasic.Core.dll
  inflating: EndpointManager/Microsoft.VisualBasic.dll
  inflating: EndpointManager/Microsoft.Win32.Primitives.dll
  inflating: EndpointManager/Microsoft.Win32.Registry.dll
  inflating: EndpointManager/Microsoft.Win32.SystemEvents.dll
  inflating: EndpointManager/mscorlib.dll
  inflating: EndpointManager/netstandard.dll
  inflating: EndpointManager/Newtonsoft.Json.Bson.dll
  inflating: EndpointManager/Newtonsoft.Json.dll
  inflating: EndpointManager/SOS_README.md
  inflating: EndpointManager/System.AppContext.dll
  inflating: EndpointManager/System.Buffers.dll
  inflating: EndpointManager/System.Collections.Concurrent.dll
  inflating: EndpointManager/System.Collections.dll
  inflating: EndpointManager/System.Collections.Immutable.dll
  inflating: EndpointManager/System.Collections.NonGeneric.dll
  inflating: EndpointManager/System.Collections.Specialized.dll
  inflating: EndpointManager/System.ComponentModel.Annotations.dll
  inflating: EndpointManager/System.ComponentModel.DataAnnotations.dll
  inflating: EndpointManager/System.ComponentModel.dll
  inflating: EndpointManager/System.ComponentModel.EventBasedAsync.dll
  inflating: EndpointManager/System.ComponentModel.Primitives.dll
  inflating: EndpointManager/System.ComponentModel.TypeConverter.dll
  inflating: EndpointManager/System.Configuration.dll
  inflating: EndpointManager/System.Console.dll
  inflating: EndpointManager/System.Core.dll
  inflating: EndpointManager/System.Data.Common.dll
  inflating: EndpointManager/System.Data.DataSetExtensions.dll
  inflating: EndpointManager/System.Data.dll
  inflating: EndpointManager/System.Diagnostics.Contracts.dll
  inflating: EndpointManager/System.Diagnostics.Debug.dll
  inflating: EndpointManager/System.Diagnostics.DiagnosticSource.dll
  inflating: EndpointManager/System.Diagnostics.EventLog.dll
  inflating: EndpointManager/System.Diagnostics.FileVersionInfo.dll
  inflating: EndpointManager/System.Diagnostics.Process.dll
  inflating: EndpointManager/System.Diagnostics.StackTrace.dll
  inflating: EndpointManager/System.Diagnostics.TextWriterTraceListener.dll
  inflating: EndpointManager/System.Diagnostics.Tools.dll
  inflating: EndpointManager/System.Diagnostics.TraceSource.dll
  inflating: EndpointManager/System.Diagnostics.Tracing.dll
  inflating: EndpointManager/System.dll
  inflating: EndpointManager/System.Drawing.Common.dll
  inflating: EndpointManager/System.Drawing.dll
  inflating: EndpointManager/System.Drawing.Primitives.dll
  inflating: EndpointManager/System.Dynamic.Runtime.dll
  inflating: EndpointManager/System.Globalization.Calendars.dll
  inflating: EndpointManager/System.Globalization.dll
  inflating: EndpointManager/System.Globalization.Extensions.dll
  inflating: EndpointManager/System.Globalization.Native.dylib
  inflating: EndpointManager/System.IdentityModel.Tokens.Jwt.dll
  inflating: EndpointManager/System.IO.Compression.Brotli.dll
  inflating: EndpointManager/System.IO.Compression.dll
  inflating: EndpointManager/System.IO.Compression.FileSystem.dll
  inflating: EndpointManager/System.IO.Compression.Native.a
  inflating: EndpointManager/System.IO.Compression.Native.dylib
  inflating: EndpointManager/System.IO.Compression.ZipFile.dll
  inflating: EndpointManager/System.IO.dll
  inflating: EndpointManager/System.IO.FileSystem.AccessControl.dll
  inflating: EndpointManager/System.IO.FileSystem.dll
  inflating: EndpointManager/System.IO.FileSystem.DriveInfo.dll
  inflating: EndpointManager/System.IO.FileSystem.Primitives.dll
  inflating: EndpointManager/System.IO.FileSystem.Watcher.dll
  inflating: EndpointManager/System.IO.IsolatedStorage.dll
  inflating: EndpointManager/System.IO.MemoryMappedFiles.dll
  inflating: EndpointManager/System.IO.Pipelines.dll
  inflating: EndpointManager/System.IO.Pipes.AccessControl.dll
  inflating: EndpointManager/System.IO.Pipes.dll
  inflating: EndpointManager/System.IO.UnmanagedMemoryStream.dll
  inflating: EndpointManager/System.Linq.dll
  inflating: EndpointManager/System.Linq.Expressions.dll
  inflating: EndpointManager/System.Linq.Parallel.dll
  inflating: EndpointManager/System.Linq.Queryable.dll
  inflating: EndpointManager/System.Memory.dll
  inflating: EndpointManager/System.Native.a
  inflating: EndpointManager/System.Native.dylib
  inflating: EndpointManager/System.Net.dll
  inflating: EndpointManager/System.Net.Http.dll
  inflating: EndpointManager/System.Net.Http.Native.a
  inflating: EndpointManager/System.Net.Http.Native.dylib
  inflating: EndpointManager/System.Net.HttpListener.dll
  inflating: EndpointManager/System.Net.Mail.dll
  inflating: EndpointManager/System.Net.NameResolution.dll
  inflating: EndpointManager/System.Net.NetworkInformation.dll
  inflating: EndpointManager/System.Net.Ping.dll
  inflating: EndpointManager/System.Net.Primitives.dll
  inflating: EndpointManager/System.Net.Requests.dll
  inflating: EndpointManager/System.Net.Security.dll
  inflating: EndpointManager/System.Net.Security.Native.a
  inflating: EndpointManager/System.Net.Security.Native.dylib
  inflating: EndpointManager/System.Net.ServicePoint.dll
  inflating: EndpointManager/System.Net.Sockets.dll
  inflating: EndpointManager/System.Net.WebClient.dll
  inflating: EndpointManager/System.Net.WebHeaderCollection.dll
  inflating: EndpointManager/System.Net.WebProxy.dll
  inflating: EndpointManager/System.Net.WebSockets.Client.dll
  inflating: EndpointManager/System.Net.WebSockets.dll
  inflating: EndpointManager/System.Numerics.dll
  inflating: EndpointManager/System.Numerics.Vectors.dll
  inflating: EndpointManager/System.ObjectModel.dll
  inflating: EndpointManager/System.Private.CoreLib.dll
  inflating: EndpointManager/System.Private.DataContractSerialization.dll
  error:  invalid compressed data to inflate
  inflating: EndpointManager/System.Private.Uri.dll
  inflating: EndpointManager/System.Private.Xml.dll
  inflating: EndpointManager/System.Private.Xml.Linq.dll
  inflating: EndpointManager/System.Reflection.DispatchProxy.dll
  inflating: EndpointManager/System.Reflection.dll
  inflating: EndpointManager/System.Reflection.Emit.dll
  inflating: EndpointManager/System.Reflection.Emit.ILGeneration.dll
  inflating: EndpointManager/System.Reflection.Emit.Lightweight.dll
  inflating: EndpointManager/System.Reflection.Extensions.dll
  inflating: EndpointManager/System.Reflection.Metadata.dll
  inflating: EndpointManager/System.Reflection.Primitives.dll
  inflating: EndpointManager/System.Reflection.TypeExtensions.dll
  inflating: EndpointManager/System.Resources.Reader.dll
  inflating: EndpointManager/System.Resources.ResourceManager.dll
  inflating: EndpointManager/System.Resources.Writer.dll
  inflating: EndpointManager/System.Runtime.CompilerServices.Unsafe.dll
  inflating: EndpointManager/System.Runtime.CompilerServices.VisualC.dll
  inflating: EndpointManager/System.Runtime.dll
  inflating: EndpointManager/System.Runtime.Extensions.dll
  inflating: EndpointManager/System.Runtime.Handles.dll
  inflating: EndpointManager/System.Runtime.InteropServices.dll
  inflating: EndpointManager/System.Runtime.InteropServices.RuntimeInformation.dll
  inflating: EndpointManager/System.Runtime.InteropServices.WindowsRuntime.dll
  inflating: EndpointManager/System.Runtime.Intrinsics.dll
  inflating: EndpointManager/System.Runtime.Loader.dll
  inflating: EndpointManager/System.Runtime.Numerics.dll
  inflating: EndpointManager/System.Runtime.Serialization.dll
  inflating: EndpointManager/System.Runtime.Serialization.Formatters.dll
  inflating: EndpointManager/System.Runtime.Serialization.Json.dll
  inflating: EndpointManager/System.Runtime.Serialization.Primitives.dll
  inflating: EndpointManager/System.Runtime.Serialization.Xml.dll
  inflating: EndpointManager/System.Runtime.WindowsRuntime.dll
  inflating: EndpointManager/System.Runtime.WindowsRuntime.UI.Xaml.dll
  inflating: EndpointManager/System.Security.AccessControl.dll
  inflating: EndpointManager/System.Security.Claims.dll
  inflating: EndpointManager/System.Security.Cryptography.Algorithms.dll
  inflating: EndpointManager/System.Security.Cryptography.Cng.dll
  inflating: EndpointManager/System.Security.Cryptography.Csp.dll
  inflating: EndpointManager/System.Security.Cryptography.Encoding.dll
  inflating: EndpointManager/System.Security.Cryptography.Native.Apple.a
  inflating: EndpointManager/System.Security.Cryptography.Native.Apple.dylib
  inflating: EndpointManager/System.Security.Cryptography.Native.OpenSsl.a
  inflating: EndpointManager/System.Security.Cryptography.Native.OpenSsl.dylib
  inflating: EndpointManager/System.Security.Cryptography.OpenSsl.dll
  inflating: EndpointManager/System.Security.Cryptography.Pkcs.dll
  inflating: EndpointManager/System.Security.Cryptography.Primitives.dll
  inflating: EndpointManager/System.Security.Cryptography.X509Certificates.dll
  inflating: EndpointManager/System.Security.Cryptography.Xml.dll
  inflating: EndpointManager/System.Security.dll
  inflating: EndpointManager/System.Security.Permissions.dll
  inflating: EndpointManager/System.Security.Principal.dll
  inflating: EndpointManager/System.Security.Principal.Windows.dll
  inflating: EndpointManager/System.Security.SecureString.dll
  inflating: EndpointManager/System.ServiceModel.Web.dll
  inflating: EndpointManager/System.ServiceProcess.dll
  inflating: EndpointManager/System.ServiceProcess.ServiceController.dll
  inflating: EndpointManager/System.Text.Encoding.CodePages.dll
  inflating: EndpointManager/System.Text.Encoding.dll
  inflating: EndpointManager/System.Text.Encoding.Extensions.dll
  inflating: EndpointManager/System.Text.Encodings.Web.dll
  inflating: EndpointManager/System.Text.Json.dll
  inflating: EndpointManager/System.Text.RegularExpressions.dll
  inflating: EndpointManager/System.Threading.Channels.dll
  inflating: EndpointManager/System.Threading.dll
  inflating: EndpointManager/System.Threading.Overlapped.dll
  inflating: EndpointManager/System.Threading.Tasks.Dataflow.dll
  inflating: EndpointManager/System.Threading.Tasks.dll
  inflating: EndpointManager/System.Threading.Tasks.Extensions.dll
  inflating: EndpointManager/System.Threading.Tasks.Parallel.dll
  inflating: EndpointManager/System.Threading.Thread.dll
  inflating: EndpointManager/System.Threading.ThreadPool.dll
  inflating: EndpointManager/System.Threading.Timer.dll
  inflating: EndpointManager/System.Transactions.dll
  inflating: EndpointManager/System.Transactions.Local.dll
  inflating: EndpointManager/System.ValueTuple.dll
  inflating: EndpointManager/System.Web.dll
  inflating: EndpointManager/System.Web.HttpUtility.dll
  inflating: EndpointManager/System.Windows.dll
  inflating: EndpointManager/System.Windows.Extensions.dll
  inflating: EndpointManager/System.Xml.dll
  inflating: EndpointManager/System.Xml.Linq.dll
  inflating: EndpointManager/System.Xml.ReaderWriter.dll
  inflating: EndpointManager/System.Xml.Serialization.dll
  inflating: EndpointManager/System.Xml.XDocument.dll
  inflating: EndpointManager/System.Xml.XmlDocument.dll
  inflating: EndpointManager/System.Xml.XmlSerializer.dll
  inflating: EndpointManager/System.Xml.XPath.dll
  inflating: EndpointManager/System.Xml.XPath.XDocument.dll
  inflating: EndpointManager/WindowsBase.dll
  inflating: EndpointManager/YamlDotNet.dll
  inflating: Fractions.dll
  inflating: kubectl/osx/kubectl
  inflating: KubernetesClient.dll
  inflating: libclrjit.dylib
  inflating: libcoreclr.dylib
  inflating: libdbgshim.dylib
  inflating: libhostfxr.dylib
  inflating: libhostpolicy.dylib
  inflating: libmscordaccore.dylib
  inflating: libmscordbi.dylib
  inflating: license.rtf
  inflating: MessagePack.dll
  inflating: Microsoft.ApplicationInsights.dll
  inflating: Microsoft.AspNetCore.Antiforgery.dll
  inflating: Microsoft.AspNetCore.Authentication.Abstractions.dll
  inflating: Microsoft.AspNetCore.Authentication.Core.dll
  inflating: Microsoft.AspNetCore.Authorization.dll
  inflating: Microsoft.AspNetCore.Authorization.Policy.dll
  inflating: Microsoft.AspNetCore.Connections.Abstractions.dll
  inflating: Microsoft.AspNetCore.Cors.dll
  inflating: Microsoft.AspNetCore.Cryptography.Internal.dll
  inflating: Microsoft.AspNetCore.DataProtection.Abstractions.dll
  inflating: Microsoft.AspNetCore.DataProtection.dll
  inflating: Microsoft.AspNetCore.Diagnostics.Abstractions.dll
  inflating: Microsoft.AspNetCore.Hosting.Abstractions.dll
  inflating: Microsoft.AspNetCore.Hosting.dll
  inflating: Microsoft.AspNetCore.Hosting.Server.Abstractions.dll
  inflating: Microsoft.AspNetCore.Html.Abstractions.dll
  inflating: Microsoft.AspNetCore.Http.Abstractions.dll
  inflating: Microsoft.AspNetCore.Http.Connections.Client.dll
  inflating: Microsoft.AspNetCore.Http.Connections.Common.dll
  inflating: Microsoft.AspNetCore.Http.dll
  inflating: Microsoft.AspNetCore.Http.Extensions.dll
  inflating: Microsoft.AspNetCore.Http.Features.dll
  inflating: Microsoft.AspNetCore.JsonPatch.dll
  inflating: Microsoft.AspNetCore.Localization.dll
  inflating: Microsoft.AspNetCore.Mvc.Abstractions.dll
  inflating: Microsoft.AspNetCore.Mvc.ApiExplorer.dll
  inflating: Microsoft.AspNetCore.Mvc.Core.dll
  inflating: Microsoft.AspNetCore.Mvc.Cors.dll
  inflating: Microsoft.AspNetCore.Mvc.DataAnnotations.dll
  inflating: Microsoft.AspNetCore.Mvc.dll
  inflating: Microsoft.AspNetCore.Mvc.Formatters.Json.dll
  inflating: Microsoft.AspNetCore.Mvc.Localization.dll
  inflating: Microsoft.AspNetCore.Mvc.Razor.dll
  inflating: Microsoft.AspNetCore.Mvc.Razor.Extensions.dll
  inflating: Microsoft.AspNetCore.Mvc.RazorPages.dll
  inflating: Microsoft.AspNetCore.Mvc.TagHelpers.dll
  inflating: Microsoft.AspNetCore.Mvc.ViewFeatures.dll
  inflating: Microsoft.AspNetCore.Razor.dll
  inflating: Microsoft.AspNetCore.Razor.Language.dll
  inflating: Microsoft.AspNetCore.Razor.Runtime.dll
  inflating: Microsoft.AspNetCore.ResponseCaching.Abstractions.dll
  inflating: Microsoft.AspNetCore.Routing.Abstractions.dll
  inflating: Microsoft.AspNetCore.Routing.dll
  inflating: Microsoft.AspNetCore.Server.Kestrel.Core.dll
  inflating: Microsoft.AspNetCore.Server.Kestrel.dll
  inflating: Microsoft.AspNetCore.Server.Kestrel.Https.dll
  inflating: Microsoft.AspNetCore.Server.Kestrel.Transport.Abstractions.dll
  inflating: Microsoft.AspNetCore.Server.Kestrel.Transport.Sockets.dll
  inflating: Microsoft.AspNetCore.SignalR.Client.Core.dll
  inflating: Microsoft.AspNetCore.SignalR.Client.dll
  inflating: Microsoft.AspNetCore.SignalR.Common.dll
  inflating: Microsoft.AspNetCore.SignalR.Protocols.Json.dll
  inflating: Microsoft.AspNetCore.SignalR.Protocols.MessagePack.dll
  inflating: Microsoft.AspNetCore.WebUtilities.dll
  inflating: Microsoft.Azure.KeyVault.dll
  inflating: Microsoft.Azure.KeyVault.WebKey.dll
  inflating: Microsoft.Azure.Management.AppService.Fluent.dll
  inflating: Microsoft.Azure.Management.BatchAI.Fluent.dll
  inflating: Microsoft.Azure.Management.Cdn.Fluent.dll
  inflating: Microsoft.Azure.Management.Compute.Fluent.dll
  inflating: Microsoft.Azure.Management.ContainerInstance.Fluent.dll
  inflating: Microsoft.Azure.Management.ContainerRegistry.Fluent.dll
  inflating: Microsoft.Azure.Management.ContainerService.Fluent.dll
  inflating: Microsoft.Azure.Management.CosmosDB.Fluent.dll
  inflating: Microsoft.Azure.Management.Dns.Fluent.dll
  inflating: Microsoft.Azure.Management.EventHub.Fluent.dll
  inflating: Microsoft.Azure.Management.Fluent.dll
  inflating: Microsoft.Azure.Management.Graph.RBAC.Fluent.dll
  inflating: Microsoft.Azure.Management.KeyVault.Fluent.dll
  inflating: Microsoft.Azure.Management.Locks.Fluent.dll
  inflating: Microsoft.Azure.Management.Monitor.Fluent.dll
  inflating: Microsoft.Azure.Management.Msi.Fluent.dll
  error:  invalid compressed data to inflate
file #458:  bad zipfile offset (local header sig):  73530626
  inflating: Microsoft.Azure.Management.PrivateDns.Fluent.dll
  inflating: Microsoft.Azure.Management.Redis.Fluent.dll
  inflating: Microsoft.Azure.Management.ResourceManager.Fluent.dll
  inflating: Microsoft.Azure.Management.Search.Fluent.dll
  inflating: Microsoft.Azure.Management.ServiceBus.Fluent.dll
  inflating: Microsoft.Azure.Management.Sql.Fluent.dll
  inflating: Microsoft.Azure.Management.Storage.Fluent.dll
  inflating: Microsoft.Azure.Management.TrafficManager.Fluent.dll
  inflating: Microsoft.CodeAnalysis.CSharp.dll
  inflating: Microsoft.CodeAnalysis.dll
  inflating: Microsoft.CodeAnalysis.Razor.dll
  inflating: Microsoft.CSharp.dll
  inflating: Microsoft.DevSpaces.Common.dll
  inflating: Microsoft.DevSpaces.Common.Json.dll
  inflating: Microsoft.DevSpaces.Library.dll
  inflating: Microsoft.DotNet.PlatformAbstractions.dll
  inflating: Microsoft.Extensions.Caching.Abstractions.dll
  inflating: Microsoft.Extensions.Caching.Memory.dll
  inflating: Microsoft.Extensions.CommandLineUtils.dll
  inflating: Microsoft.Extensions.Configuration.Abstractions.dll
  inflating: Microsoft.Extensions.Configuration.Binder.dll
  inflating: Microsoft.Extensions.Configuration.dll
  inflating: Microsoft.Extensions.Configuration.EnvironmentVariables.dll
  inflating: Microsoft.Extensions.Configuration.FileExtensions.dll
  inflating: Microsoft.Extensions.DependencyInjection.Abstractions.dll
  inflating: Microsoft.Extensions.DependencyInjection.dll
  inflating: Microsoft.Extensions.DependencyModel.dll
  inflating: Microsoft.Extensions.FileProviders.Abstractions.dll
  inflating: Microsoft.Extensions.FileProviders.Composite.dll
  inflating: Microsoft.Extensions.FileProviders.Physical.dll
  inflating: Microsoft.Extensions.FileSystemGlobbing.dll
  inflating: Microsoft.Extensions.Hosting.Abstractions.dll
  inflating: Microsoft.Extensions.Localization.Abstractions.dll
  inflating: Microsoft.Extensions.Localization.dll
  inflating: Microsoft.Extensions.Logging.Abstractions.dll
  inflating: Microsoft.Extensions.Logging.dll
  inflating: Microsoft.Extensions.ObjectPool.dll
  inflating: Microsoft.Extensions.Options.dll
  inflating: Microsoft.Extensions.Primitives.dll
  inflating: Microsoft.Extensions.WebEncoders.dll
  inflating: Microsoft.IdentityModel.Clients.ActiveDirectory.dll
  inflating: Microsoft.IdentityModel.JsonWebTokens.dll
  inflating: Microsoft.IdentityModel.Logging.dll
  inflating: Microsoft.IdentityModel.Tokens.dll
  inflating: Microsoft.Net.Http.Headers.dll
  inflating: Microsoft.Rest.ClientRuntime.Azure.Authentication.dll
  inflating: Microsoft.Rest.ClientRuntime.Azure.dll
  inflating: Microsoft.Rest.ClientRuntime.dll
  inflating: Microsoft.VisualBasic.Core.dll
  inflating: Microsoft.VisualBasic.dll
  inflating: Microsoft.Win32.Primitives.dll
  inflating: Microsoft.Win32.Registry.dll
  inflating: Microsoft.WindowsAzure.Storage.dll
  inflating: Minimatch.dll
  inflating: mscorlib.dll
  inflating: netstandard.dll
  inflating: Newtonsoft.Json.Bson.dll
  inflating: Newtonsoft.Json.dll
  inflating: SOS_README.md
  inflating: System.AppContext.dll
  inflating: System.Buffers.dll
  inflating: System.Collections.Concurrent.dll
  inflating: System.Collections.dll
  inflating: System.Collections.Immutable.dll
  inflating: System.Collections.NonGeneric.dll
  inflating: System.Collections.Specialized.dll
  inflating: System.ComponentModel.Annotations.dll
  inflating: System.ComponentModel.DataAnnotations.dll
  inflating: System.ComponentModel.dll
  inflating: System.ComponentModel.EventBasedAsync.dll
  inflating: System.ComponentModel.Primitives.dll
  inflating: System.ComponentModel.TypeConverter.dll
  inflating: System.Configuration.dll
  inflating: System.Console.dll
  inflating: System.Core.dll
  inflating: System.Data.Common.dll
  inflating: System.Data.DataSetExtensions.dll
  inflating: System.Data.dll
  inflating: System.Diagnostics.Contracts.dll
  inflating: System.Diagnostics.Debug.dll
  inflating: System.Diagnostics.DiagnosticSource.dll
  inflating: System.Diagnostics.EventLog.dll
  inflating: System.Diagnostics.FileVersionInfo.dll
  inflating: System.Diagnostics.Process.dll
  inflating: System.Diagnostics.StackTrace.dll
  inflating: System.Diagnostics.TextWriterTraceListener.dll
  inflating: System.Diagnostics.Tools.dll
  inflating: System.Diagnostics.TraceSource.dll
  inflating: System.Diagnostics.Tracing.dll
  inflating: System.dll
  inflating: System.Drawing.dll
  inflating: System.Drawing.Primitives.dll
  inflating: System.Dynamic.Runtime.dll
  inflating: System.Globalization.Calendars.dll
  inflating: System.Globalization.dll
  inflating: System.Globalization.Extensions.dll
  inflating: System.Globalization.Native.dylib
  inflating: System.IdentityModel.Tokens.Jwt.dll
  inflating: System.IO.Compression.Brotli.dll
  inflating: System.IO.Compression.dll
  inflating: System.IO.Compression.FileSystem.dll   bad CRC 53839902  (should be 1b5ca56c)
file #560:  bad zipfile offset (local header sig):  84823411
  inflating: System.IO.Compression.Native.dylib
  inflating: System.IO.Compression.ZipFile.dll
  inflating: System.IO.dll
  inflating: System.IO.FileSystem.AccessControl.dll
  inflating: System.IO.FileSystem.dll
  inflating: System.IO.FileSystem.DriveInfo.dll
  inflating: System.IO.FileSystem.Primitives.dll
  inflating: System.IO.FileSystem.Watcher.dll
  inflating: System.IO.IsolatedStorage.dll
  inflating: System.IO.MemoryMappedFiles.dll
  inflating: System.IO.Pipelines.dll
  inflating: System.IO.Pipes.AccessControl.dll
  inflating: System.IO.Pipes.dll
  inflating: System.IO.UnmanagedMemoryStream.dll
  inflating: System.Linq.dll
  inflating: System.Linq.Expressions.dll
  inflating: System.Linq.Parallel.dll
  inflating: System.Linq.Queryable.dll
  inflating: System.Memory.dll
  inflating: System.Native.a
  inflating: System.Native.dylib
  inflating: System.Net.dll
  inflating: System.Net.Http.dll
  inflating: System.Net.Http.Native.a
  inflating: System.Net.Http.Native.dylib
  inflating: System.Net.HttpListener.dll
  inflating: System.Net.Mail.dll
  inflating: System.Net.NameResolution.dll
  inflating: System.Net.NetworkInformation.dll
  inflating: System.Net.Ping.dll
  inflating: System.Net.Primitives.dll
  inflating: System.Net.Requests.dll
  error:  invalid compressed data to inflate
  inflating: System.Net.Security.dll
  inflating: System.Net.Security.Native.a
  inflating: System.Net.Security.Native.dylib
  inflating: System.Net.ServicePoint.dll
  inflating: System.Net.Sockets.dll
  inflating: System.Net.WebClient.dll
  inflating: System.Net.WebHeaderCollection.dll
  inflating: System.Net.WebProxy.dll
  inflating: System.Net.WebSockets.Client.dll
  inflating: System.Net.WebSockets.dll
  inflating: System.Numerics.dll
  inflating: System.Numerics.Vectors.dll
  inflating: System.ObjectModel.dll   bad CRC 7d46e411  (should be ea709b16)
file #606:  bad zipfile offset (local header sig):  91573823
  inflating: System.Private.DataContractSerialization.dll
  inflating: System.Private.Uri.dll
  inflating: System.Private.Xml.dll
  inflating: System.Private.Xml.Linq.dll
  inflating: System.Reflection.DispatchProxy.dll
  inflating: System.Reflection.dll
  inflating: System.Reflection.Emit.dll
  inflating: System.Reflection.Emit.ILGeneration.dll
  inflating: System.Reflection.Emit.Lightweight.dll
  inflating: System.Reflection.Extensions.dll
  inflating: System.Reflection.Metadata.dll
  inflating: System.Reflection.Primitives.dll
  inflating: System.Reflection.TypeExtensions.dll
  inflating: System.Resources.Reader.dll
  inflating: System.Resources.ResourceManager.dll
  inflating: System.Resources.Writer.dll
  inflating: System.Runtime.CompilerServices.Unsafe.dll
  inflating: System.Runtime.CompilerServices.VisualC.dll
  inflating: System.Runtime.dll
  inflating: System.Runtime.Extensions.dll
  inflating: System.Runtime.Handles.dll
  inflating: System.Runtime.InteropServices.dll
  inflating: System.Runtime.InteropServices.RuntimeInformation.dll
  inflating: System.Runtime.InteropServices.WindowsRuntime.dll
  inflating: System.Runtime.Intrinsics.dll
  inflating: System.Runtime.Loader.dll
  inflating: System.Runtime.Numerics.dll
  inflating: System.Runtime.Serialization.dll
  inflating: System.Runtime.Serialization.Formatters.dll
  inflating: System.Runtime.Serialization.Json.dll
  inflating: System.Runtime.Serialization.Primitives.dll
  inflating: System.Runtime.Serialization.Xml.dll
  inflating: System.Runtime.WindowsRuntime.dll
  inflating: System.Runtime.WindowsRuntime.UI.Xaml.dll
  inflating: System.Security.AccessControl.dll
  inflating: System.Security.Claims.dll
  inflating: System.Security.Cryptography.Algorithms.dll
  inflating: System.Security.Cryptography.Cng.dll
  inflating: System.Security.Cryptography.Csp.dll
  inflating: System.Security.Cryptography.Encoding.dll
  inflating: System.Security.Cryptography.Native.Apple.a
  inflating: System.Security.Cryptography.Native.Apple.dylib
  inflating: System.Security.Cryptography.Native.OpenSsl.a
  inflating: System.Security.Cryptography.Native.OpenSsl.dylib
  inflating: System.Security.Cryptography.OpenSsl.dll
  inflating: System.Security.Cryptography.Pkcs.dll
  inflating: System.Security.Cryptography.Primitives.dll
  inflating: System.Security.Cryptography.X509Certificates.dll
  error:  invalid compressed data to inflate
  inflating: System.Security.Cryptography.Xml.dll
  inflating: System.Security.dll
  inflating: System.Security.Permissions.dll
  inflating: System.Security.Principal.dll
  inflating: System.Security.Principal.Windows.dll
  inflating: System.Security.SecureString.dll
  inflating: System.ServiceModel.Web.dll
  inflating: System.ServiceProcess.dll
  inflating: System.ServiceProcess.ServiceController.dll
  inflating: System.Text.Encoding.CodePages.dll
  inflating: System.Text.Encoding.dll
  inflating: System.Text.Encoding.Extensions.dll
  inflating: System.Text.Encodings.Web.dll
  inflating: System.Text.Json.dll
  inflating: System.Text.RegularExpressions.dll
  inflating: System.Threading.Channels.dll
  inflating: System.Threading.dll
  inflating: System.Threading.Overlapped.dll
  inflating: System.Threading.Tasks.Dataflow.dll
  inflating: System.Threading.Tasks.dll
  inflating: System.Threading.Tasks.Extensions.dll
  inflating: System.Threading.Tasks.Parallel.dll
  inflating: System.Threading.Thread.dll
  inflating: System.Threading.ThreadPool.dll
  inflating: System.Threading.Timer.dll
  inflating: System.Transactions.dll
  inflating: System.Transactions.Local.dll
  inflating: System.ValueTuple.dll
  inflating: System.Web.dll
  inflating: System.Web.HttpUtility.dll
  inflating: System.Windows.dll
  inflating: System.Xml.dll
  inflating: System.Xml.Linq.dll
  inflating: System.Xml.ReaderWriter.dll
  inflating: System.Xml.Serialization.dll
  inflating: System.Xml.XDocument.dll
  inflating: System.Xml.XmlDocument.dll
  inflating: System.Xml.XmlSerializer.dll
  inflating: System.Xml.XPath.dll
  inflating: System.Xml.XPath.XDocument.dll
  inflating: ThirdPartyNotices.txt
  inflating: WindowsBase.dll
  inflating: YamlDotNet.dll`

**Comment by rakeshvanga** [comment]
@TinLe Thanks for trying out the manual method. Since you are working on mac, can you download the bits from this link: https://bridgetokubernetes.azureedge.net/zip/1.0.20210723.6/lpk-osx.zip and set this environment variable: `BRIDGE_BUILD_PATH=<place where you extracted the zip>`

**Comment by TinLe** [comment]
@rakeshvanga The extension installs now, but it insists on using wrong path for kubectl.  I am getting error messages similar to this popup.

`Failed to get the current context from the kubeconfig: Failed to execute: /Users/tile/Library/Application/kubectl/osx/kubectl config view --minify -o json --kubeconfig /Users/tile/.kube/config. Error: spawn /Users/tile/Library/Application/kubectl/osx/kubectl ENOENT`

I have settings.json configured to use kubectl in another location.

`cat .vscode/settings.json
{
    "vs-kubernetes": {

        "vs-kubernetes.namespace": "",
        "vs-kubernetes.kubectl-path": "/Users/tile/bin/google-cloud-sdk/bin",
        "vs-kubernetes.helm-path": "",
        "vs-kubernetes.minikube-path": "",
        "vs-kubernetes.kubectlVersioning": "user-provided",
        "vs-kubernetes.outputFormat": "yaml",
        "vs-kubernetes.kubeconfig": "",
        "vs-kubernetes.knownKubeconfigs": [],
        "vs-kubernetes.autoCleanupOnDebugTerminate": false,
        "vs-kubernetes.nodejs-autodetect-remote-root": true,
        "vs-kubernetes.nodejs-remote-root": "",
        "vs-kubernetes.nodejs-debug-port": 9229,
        "vs-kubernetes.local-tunnel-debug-provider": "",
        "checkForMinikubeUpgrade": true,
        "imageBuildTool": "Docker"
    }`

**Comment by rakeshvanga** [comment]
@TinLe There is one more step to be done to get the binaries ready on Mac.
Can you run following commands:
 `chmod +x <place where you extracted the zip>/kubectl/osx/kubectl`
`chmod +x <place where you extracted the zip>/bridge`
`chmod +x <place where you extracted the zip>/EndpointManager/EndpointManager`

**Comment by TinLe** [comment]
@rakeshvanga Yes, I did that.  The steps at the top of this bug report included making those binaries executable.

Note that `dsc` is missing from the zip package you asked me to try.   kubectl and EndpointManager is there.

**Comment by TinLe** [comment]
Interesting, I restarted VSC and am now getting the dependencies error again for this extension.

`Failed to update dependencies: Maximum number of retries exceeded. The operation 'e' failed with error: invalid distance too far back. Technical details: {"errno":-3,"code":"Z_DATA_ERROR"}`

**Comment by rakeshvanga** [comment]
`dsc` is renamed to `bridge`, can you try `chmod +x <place where you extracted the zip>/bridge`?
Also, Bridge To Kubernetes doesn't read vscode settings to get the path of the kubectl executable, it downloads its own version of kubectl.

After extracting the binaries using the environment variable `BRIDGE_BUILD_PATH`, did you set this variable to the path where you have extracted the binaries and then started the vscode session so that the vscode session picks up the env variable?

**Comment by TinLe** [comment]
> `dsc` is renamed to `bridge`, can you try `chmod +x <place where you extracted the zip>/bridge`?

I see.  Done.  Set +x on `bridge`.

> Also, Bridge To Kubernetes doesn't read vscode settings to get the path of the kubectl executable, it downloads its own version of kubectl.

Even on OS X?   There is no ~/Library/Application directory on my Mac.  I prefer to use one `kubectl` for all projects and environment to keep it simple.  The `kubectl` I use is from Google Cloud SDK.

> After extracting the binaries using the environment variable `BRIDGE_BUILD_PATH`, did you set this variable to the path where you have extracted the binaries and then started the vscode session so that the vscode session picks up the env variable?

Note that this is on OS X.   I have set the env var `BRIDGE_BUILD_PATH` in my .bashrc, restarted OS X, then start vscode.

Hmm.  It seem something is resetting that env var.  I've manually set it and started Electron manually.  Now we are getting somewhere.  I am hitting the OS X restrictions against executing d/l files not signed or from the App store.  Got past that and now getting the following errors:

`Failed to refresh the kubeconfig token: Failed to load �, error: dlopen(/Users/tile/Library/Application Support/Code/User/globalStorage/mindaro.mindaro/file-downloader-downloads/binaries/libhostfxr.dylib, 1): no suitable image found. Did find: /Users/tile/Library/Application Support/Code/User/globalStorage/mindaro.mindaro/file-downloader-downloads/binaries/libhostfxr.dylib: truncated mach-o error: segment __TEXT extends to 339968 which is past end of file 51681 /Users/tile/Library/Application Support/Code/User/globalStorage/mindaro.mindaro/file-downloader-downloads/binaries/libhostfxr.dylib: truncated mach-o error: segment __TEXT extends to 339968 which is past end of file 51681 The library libhostfxr.dylib was found, but loading it from /Users/tile/Library/Application Support/Code/User/globalStorage/mindaro.mindaro/file-downloader-downloads/binaries/libhostfxr.dylib failed - Installing .NET Core prerequisites might help resolve this problem. https://go.microsoft.com/fwlink/?...`

**Comment by rakeshvanga** [comment]
> Even on OS X? There is no ~/Library/Application directory on my Mac. I prefer to use one kubectl for all projects and environment to keep it simple. The kubectl I use is from Google Cloud SDK.

Did you extract the binaries downloaded from https://bridgetokubernetes.azureedge.net/zip/1.0.20210723.6/lpk-osx.zip to  /Users/tile/Library/Application/ as the error says in the above comment? The last section of the path `kubectl/osx/kubectl` is appended by our extension to the BRIDGE_BUILD_PATH env variable. 
`kubectl` binary is downloaded as part of the vscode extension, so that we could control the behavior of the executable otherwise every customer could be on a different version and our extension might not work.

> Failed to refresh the kubeconfig token: Failed to load �, error: dlopen(/Users/tile/Library/Application Support/Code/User/globalStorage/mindaro.mindaro/file-downloader-downloads/binaries/libhostfxr.dylib, 1): no suitable image found. Did find: /Users/tile/Library/Application Support/Code/User/globalStorage/mindaro.mindaro/file-downloader-downloads/binaries/libhostfxr.dylib:.....

I'm not sure about this error and need to look into this further. It could be related to signing issue but could you try some of the solutions provided in here: https://stackoverflow.com/questions/32730312/reason-no-suitable-image-found

**Comment by rakeshvanga** [comment]
Can you run this command `xattr -dr com.apple.quarantine` in the directory `/Users/tile/Library/Application Support/Code/User/globalStorage/mindaro.mindaro/file-downloader-downloads/binaries/` so that Mac OSX ignores this folder.

**Comment by TinLe** [comment]
> Can you run this command `xattr -dr com.apple.quarantine` in the directory `/Users/tile/Library/Application Support/Code/User/globalStorage/mindaro.mindaro/file-downloader-downloads/binaries/` so that Mac OSX ignores this folder if the issue is related to signing issue.

I did.  That's how I got past the warnings and errors :-)

**Comment by rakeshvanga** [comment]
@TinLe Sure. Were the stack overflow suggestions helpful in resolving the issue?

**Comment by TinLe** [comment]
> @TinLe Sure. Were the stack overflow suggestions helpful in resolving the issue?

Just got around to trying it.  No, stack overflow suggestions did not work for me.

**Comment by rakeshvanga** [comment]
@TinLe Thanks for trying out. We are tracking this issue internally and will update you once we find any solutions.

**Comment by MRHACKERSadmin** [comment]
> ## Issue 1: Maximum number of retries exceeded
> Some of the time, getting dependencies for Bridge to Kubernetes in VS Code may fail with this error.
> 
> ![image](https://user-images.githubusercontent.com/13313748/93949536-b6a9f480-fcf5-11ea-9d8e-fc6749da17de.png) If the issue reproduces consistently after hitting the retry button, a manual workaround is the best way to proceed.
> 
> **The workaround**
> 
> 1. Visit one of the below links in a browser (choose based on your platform):
>    https://bridgetokubernetes.blob.core.windows.net/zip/lks/lpk-win.zip
>    https://bridgetokubernetes.blob.core.windows.net/zip/lks/lpk-osx.zip
>    https://bridgetokubernetes.blob.core.windows.net/zip/lks/lpk-linux.zip
>    This will prompt the download of a .zip file.
> 2. Unzip the file and move the contents to the Visual Studio Code extension storage folder.  You may need to create the directory if it does not exist.
>    On Windows: %UserProfile%\AppData\Roaming\Code\User\globalStorage\mindaro.mindaro\file-downloader-downloads\binaries
>    On MacOS: "/Users/**your_username**/Library/Application Support/Code/User/globalStorage/mindaro.mindaro/file-downloader-downloads/binaries"
>    On Linux: /home/**your_username**/.config/Code/User/globalStorage/mindaro.mindaro/file-downloader-downloads/binaries
> 3. To check that the hierarchy is correct, you should now see folders called "kubectl" and "EndpointManager" along with several other files side-by-side under the /binaries/ directory.
> 4. IMPORTANT: If you are on Linux or Mac, you need to run: `sudo chmod +x` on dsc, kubectl/**your_platform**/kubectl and EndpointManager/EndpointManager.
> 5. Open VS Code again.  The extension should now be able to find and execute its dependencies.
> 
> **Long-term** We are currently at work trying to understand why this error happens and to improve the experience! Thank you for your patience.
> 
> ## Issue 2: EACCESS error
> If you are on Linux, you might run into this issue:
> 
> ![image](https://user-images.githubusercontent.com/13313748/94971130-0b4e2c00-04bb-11eb-8a19-61be35ab7607.png)
> 
> **The workaround**
> 
> 1. From a terminal, navigate to /home/**your_username**/.config/Code/User/globalStorage/mindaro.mindaro/file-downloader-downloads/binaries
> 2. Run: `sudo chmod +x` on dsc, kubectl/linux/kubectl and EndpointManager/EndpointManager.
> 3. Open VS Code again.  The extension should now be able to find and execute its dependencies.
> 
> **Long-term** This issue will be fixed in our next release -- stay tuned.

Duplicate of #

**Comment by MHacker9404** [comment]
I'm getting this variation:
Failed to update dependencies: The "streams" argument must be specified
I've attached the log file...
[mindaro-vscode-2022-03-23T11-15-12.197Z.txt](https://github.com/microsoft/mindaro/files/8332561/mindaro-vscode-2022-03-23T11-15-12.197Z.txt)
.

**Comment by coxantohi** [comment]
The link from Step 1, Issue 1 is no longer valid.
I had the same error (Failed to update dependencies: The "streams" argument must be specified) when unzipping that file to the binaries folder.

To get the correct download link you must use the "Developer: Open Extensions Logs Folder" command to open the folder where the "Kubernetes" extension logs reside. For me it was C:\Users\%username%\AppData\Roaming\Code\logs\20220401T164522\exthost1

There, inside the "mindaro.mindaro" folder you will find a a log file. Mine was "mindaro-vscode-2022-04-01T13-45-26.503Z.txt"

You should find a line having something like: "Resolved versionUrl 'https://bridgetokubernetes.blob.core.windows.net/zip/1.0.20220125.5/lks.json' <json>{}</json>" If you open that link in a browser you will find that the url is slightly different. Mine was https://bridgetokubernetes.azureedge.net/zip/1.0.20220125.5/lpk-win.zip. This file is different than https://bridgetokubernetes.blob.core.windows.net/zip/lks/lpk-win.zip, and unzipping it in the "binaries" folder as mentioned in the instructions worked.

**Comment by almostsultry** [comment]
I also have the "streams" error on osx. I followed your instructions and downloaded the new .zip for osx and then followed the instructions to place the files in the binaries folder and chmod +x the 3 files. I still have the same error. Where do I find the log file on osx to submit for assistance?

**Comment by Shady6** [comment]
Hello,
Thank you @coxantohi for providing working asnwer. 

@amsoedal or other devs could you please update urls from the first answer (I mean these urls looking like this one https://bridgetokubernetes.blob.core.windows.net/zip/lks/lpk-linux.zip) so they point to the newest version of binaries? The problem is that they have some outdated stuff like there is 'dsc' but no 'bridge'.

As for time of writing this message correct urls are returned by this url https://bridgetokubernetes.blob.core.windows.net/zip/1.0.20220125.5/lks.json

**Comment by kfimchenko** [comment]
Unzipped files from https://bridgetokubernetes.azureedge.net/zip/1.0.20220125.5/lpk-win.zip to PATH folder, so bridge command works. But the error still happening
![image](https://user-images.githubusercontent.com/7370396/162803750-b1599543-012f-441a-be36-979c8b8d3df1.png)

[mindaro-vscode-2022-04-11T18-08-51.160Z.txt](https://github.com/microsoft/mindaro/files/8466754/mindaro-vscode-2022-04-11T18-08-51.160Z.txt)

Are there any ideas on what can I do to fix it?

**Comment by eduborgono** [comment]
The solution for the first issue is not updated. I followed all the steps and the error is still happening, also followed the suggestions of all the nice people of the comments and the error is still there.
Besides, I should comment that the json [lks.json](https://bridgetokubernetes.blob.core.windows.net/zipv2/1.0.20220125.5/lks.json) has changed because it doesn't even have the same structure than before neither the links look like before, the `zip` path segment changed to `zipv2`.
I don't think the change from `zip` to `zipv2` has anything to do with the solution, however the change of [lks.json](https://bridgetokubernetes.blob.core.windows.net/zipv2/1.0.20220125.5/lks.json) lead me to think that the steps that imply download, uncompress & move files changed in the steps of the solution.

**Comment by MooDySmiles** [comment]
I'm in the same situation as @coxantohi with the `Failed to update dependencies: The "streams" argument must be specified`, but downloading the zip from the URL `https://bridgetokubernetes.azureedge.net/zip/1.0.20220125.5/lpk-linux.zip` (since I'm on linux) didn't do the trick for me, I'm still getting the same error over and over again.

**Comment by MooDySmiles** [comment]
Ok back on my words. Solution proposed by @coxantohi worked for me too, the only difference was that I did not have a `binaries` folder inside `/home/<my-user>/.config/Code/User/globalStorage/mindaro.mindaro/file-downloader-downloads/`, it was called `bridge`.

Putting there the files from [https://bridgetokubernetes.azureedge.net/zip/1.0.20220125.5/lpk-linux.zip](https://bridgetokubernetes.azureedge.net/zip/1.0.20220125.5/lpk-linux.zip) and renaming the folder to `binaries` **did** the trick for me.

Why the folder was named like that? Dunno, the important thing now is that **_IT WORKS_**

**Comment by jackliusr** [comment]
I got "Failed to update dependencies: The "streams" argument must be specified. "  when run in WSL , I downloaded "https://bridgetokubernetes.azureedge.net/zip/1.0.20220125.5/lpk-linux.zip" in https://bridgetokubernetes.blob.core.windows.net/zip/1.0.20220125.5/lks.json.  and do the "chmod +x bridge, kubectl/linux/kubectl and EndpointManager/EndpointManager" 
[mindaro-vscode-2022-04-23T16-05-47.226Z.txt](https://github.com/microsoft/mindaro/files/8547579/mindaro-vscode-2022-04-23T16-05-47.226Z.txt)

**Comment by mchaker** [comment]
@jackliusr 's solution worked for me as well. This is on debian via crostini (on chromeos)

**Comment by PabloZaiden** [comment]
How to have *Bridge to Kubernetes* up and running on MacOS:

- As of today, use the following link to get the binaries for your platform: https://bridgetokubernetes.blob.core.windows.net/zip/1.0.20220125.5/lks.json
- Unzip the binaries in the `/Users/your_username/Library/Application Support/Code/User/globalStorage/mindaro.mindaro/file-downloader-downloads/binaries` directory. Make sure there is no top-level directory containing all the files in the zip.
- Run the following command to add execution permissions: `chmod +x bridge kubectl/osx/kubectl  EndpointManager/EndpointManager`
- Reopen VSCode. If the binaries are not signed and you still want to execute them (not recommended, unless you know what you're doing), you will have to go to System Preferences, Security & Privacy and allow the execution of those binaries. All the binaries.
- Reopen VSCode again. Now it should work.

Again. Avoid executing unsigned binaries.

**Comment by daniv-msft** [comment]
We released a fix to the [File Downloader](https://github.com/microsoft/vscode-file-downloader) extension we have a dependency on to download. This should fix the `"streams" argument must be specified.` error. Thanks everyone!

**Comment by jmclong** [comment]
Thank you! I can confirm that I'm no longer seeing the `streams` error, and things seem to be working properly.
